src/transform/globalize.js

```diff
@@ -29,30 +28,93 @@ function indent (code) {
   return lines.join('\n');
 }
 
-function relative (file) {
+function makePathRelative (file) {
   return path.relative(process.cwd(), file);
 }
 
+function defineDependencies (imports, dependencies) {
+  var code = '';
+  var keyVals = imports.map(function (imp, idx) {
+    return '"' + imp + '": ' + generateModuleName(dependencies[idx]);
+  });
+
+  keyVals.unshift('"exports": exports');
+  keyVals.unshift('"module": module');
+  code = '\n' + indent(keyVals.join(',\n')) + '\n';
+  return '{' + code + '}';
+}
+
+function defineReplacement (name, deps, func) {
+  function overload (type) {
```

Comment:
The code to overload amd arguments would be about this much anyways. Was hoping to abstract it out at some point. Maybe this could be made into a more concise block instead. Thoughts?